glean-core/src/metrics/dual_labeled_counter.rs

```diff
@@ -0,0 +1,375 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use std::borrow::Cow;
+use std::char;
+use std::collections::HashMap;
+use std::io::Read;
+use std::sync::{Arc, Mutex};
+
+use crate::common_metric_data::{CommonMetricData, CommonMetricDataInternal, DynamicLabelType};
+use crate::error_recording::{record_error, test_get_num_recorded_errors, ErrorType};
+use crate::metrics::{CounterMetric, Metric, MetricType};
+use crate::Glean;
+
+const MAX_LABELS: usize = 16;
+const OTHER_LABEL: &str = "__other__";
+const MAX_LABEL_LENGTH: usize = 111;
+const RECORD_SEPARATOR: char = '\x1E';
+
+/// A dual labled metric, mirrors the Legacy keyed-categorical type.
```

Comment:
We shouldn't refer to the legacy type here. Glean doesn't know anything about Legacy.